components/org.wso2.carbon.identity.application.authz.xacml/src/main/java/org/wso2/carbon/identity/application/authz/xacml/handler/impl/XACMLBasedAuthorizationHandler.java

```diff
@@ -103,6 +104,10 @@ public boolean isAuthorized(HttpServletRequest request, HttpServletResponse resp
                 if (log.isDebugEnabled()) {
                     log.debug("XACML Authorization request :\n" + requestString);
                 }
+
+                PrivilegedCarbonContext.startTenantFlow();
```

Comment:
We had a util method to start a tenant flow IIRC. Please check whether you can reuse it.